### PR TITLE
Fix bug in `numberOfIntegerDigits` calculation, simplifying the algorithm

### PR DESCRIPTION
**Pull request by mosuem**
This fixes a bug where the number of digits may be wrong due to rounding errors.

### CODE REVIEW

File: lib/src/intl/number_format.dart [hunk 1]
```diff
@@ -618,9 +618,9 @@ class NumberFormat {
     if (simpleNumber < 100000000000000) return 14;
     if (simpleNumber < 1000000000000000) return 15;
     if (simpleNumber < 10000000000000000) return 16;
-    // We're past the point where being off by one on the number of digits
-    // will affect the pattern, so now we can use logs.
-    return max(1, (log(simpleNumber) / _ln10).ceil());
+    if (simpleNumber < 100000000000000000) return 17;
+    if (simpleNumber < 1000000000000000000) return 18;
+    return 19;
```
Comment (devoncarew) [inline lib/src/intl/number_format.dart:623]: I assume we're not expecting anything past 19?

File: test/number_format_test.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import 'package:intl/intl.dart';
```
Comment (devoncarew) [inline test/number_format_test.dart:1]: nit: add a copyright header for this file

**Review by devoncarew — APPROVED** [review]
lgtm w/ the indicated change related to the file header

**Review by patel-aum — APPROVED** [review]

File: lib/src/intl/number_format.dart (hunk 1 above)
Comment (mosuem) [reply lib/src/intl/number_format.dart:623]: Anything larger cannot be represented in 64 bits, so we are safe!

**Review by devoncarew — APPROVED** [review]

**Pull request merged**